src/julia_to_gap.jl

```diff
@@ -175,4 +189,33 @@ function julia_to_gap(
     return record
 end
 
+## GAP objects: deal with recursion (subobjects may be Julia objects)
+function julia_to_gap(
+    obj::GapObj,
+    recursion_dict::IdDict{Any,Any} = IdDict();
+    recursive::Bool = false,
+)
```

Comment:
By chance I was working on changes to the `julia_to_gap` and `gap_to_julia` code in the past couple days as well, and I started to really wonder about the way we set it up. Like, we keep having to repeat these default arguments. 

I think it would make more sense if we refactored both of these functions, by changing *all* methods that actually do something to take the full set of arguments (without default arguments specified); and then have one set of "generic" functions at the top level which add in the default values for arguments if missing. This way we safe we are guaranteed to get consistent defaults.

Actually, I might go a bit further:
- I'd rename the "worker methods" for `julia_to_gap` to `_ julia_to_gap`, and change them all to have *two* arguments: `obj` and `recursion_dict`, where the latter is allow to be `nothing` (so type would be `Union{Nothing, IdDict{Any,Any}}`. Then we don't need to allocate an IdDict that we don't end up using (but perhaps Julia already is clever enough to do that for us?)
- all worker methods would then only invoke `_ julia_to_gap` but never `julia_to_gap`, which is reserved for clients
- the new function `julia_to_gap` would then look like this:
```julia
julia_to_gap(obj; recursive::Bool = true) = _julia_to_gap(obj, recursive ? IdDict() : nothing)
```
- similar for `gap_to_julia`; except we already have a `_gap_to_julia`; but I'd just remove that, and instead teach the high level `gap_to_julia` (and perhaps also `julia_to_gap`) to *also* accept recursive as a positional argument:
```julia
julia_to_gap(obj, recursive) = _julia_to_gap(obj, recursive ? IdDict() : nothing)
```

Thoughts?

Reply:
This sounds reasonable.
(For programmatic access such as via `convert`, we have to support requests for recursive conversions also for objects which do not have subobjects, and some cases were missing.
Your proposal is more conceptual from this viewpoint.)
Do you have already code for these changes?